src/app/data-client.ts

```diff
@@ -53,6 +69,68 @@ export class DataClient {
     this.dataSyncClient = createPromiseClient(DataSyncService, transport);
   }
 
+  /**
+   * Obtain unified tabular data and metadata from the specified data source.
+   *
+   * @param partId The ID of the part that owns the data
+   * @param resourceName The name of the requested resource that captured the
+   *   data
+   * @param resourceSubtype The subtype of the requested resource that captured
+   *   the data
+   * @param methodName The data capture method name
+   * @param startTime Optional start time (`Date` object) for requesting a specific range of data
+   * @param endTime Optional end time (`Date` object) for requesting a specific range of data
+   * @returns An array of unified tabular data and metadata.
+   */
+  async exportTabularData(
+    partId: string,
+    resourceName: string,
+    resourceSubtype: string,
+    methodName: string,
+    startTime?: Date,
+    endTime?: Date
+  ) {
```

Comment:
@stuqdog Wondering if I should change this to return `AsyncIterableIterator<TabularDataPoint>`. Thoughts?

Reply:
Hmm... I suspect we need to be `await`ing ourselves either way to convert into a `TabularDataPoint` (or get into some funky async functional mapping). I guess that might be useful if someone just wants to stream all tabular data as it continues to come in? That seems like it adds more complexity/work to users so my thought is no, but there may be a use case I'm not thinking about (could someone use that to tail all data as it comes in? is that a use case we expect to be common?).

tl;dr weakly I support keeping as-is but open to being convinced otherwise.

Reply:
I'm thinking of the use case where someone wants to retrieve a lot of data. Possibly even download it as a file